WordPress/gutenberg
# Components: Use `Element.scrollIntoView()` instead of `dom-scroll-into-view`

## What?
This PR suggests that we utilize the native `Element.scrollIntoView()` instead of the external package `dom-scroll-into-view`.

## Why?
All things being equal, using native functionality is always preferred to using external packages and libraries.

But also, the use case for `dom-scroll-into-view` is quite straightforward and goes way back to the initial implementation of `FormTokenField` in https://github.com/WordPress/gutenberg/pull/924 by @youknowriad. 

From what I understand, there's no good reason to use the external package. `Element.scrollIntoView()` has a solid browser support already and works really well in testing. 

## How?
* Replacing the `dom-scroll-into-view` syntax with `Element.scrollIntoView()` calls in both occurrences:
  * `URLInput`
  * `FormTokenField`
* Removing the unnecessary state that maintains whether we're currently scrolling - I couldn't spot a good reason to keep it, and I don't see a difference in testing.
* Removing the `dom-scroll-into-view` dependency, along with typings and a related lerna patch.

## Testing Instructions
* Make sure you have at least 10 pages that start with the same name ("Page 1", "Page 2", ... "Page 10")
* Make sure you have at least 10 tags that start with the same name ("Test 1", "Test 2", ... "Test 10"), and each of those tags is assigned to at least one post.
* Go to the post editor.
* Insert some text in a paragraph, select it, click on the anchor to make it a link, input "Page" in the search field, and verify that as you use the arrows, the suggestions are scrolled so the selected result is always visible.
* Head to the sidebar, and reveal the Tags box.
* In the tag field, input "Test" in the search field, and verify that as you use the arrows, the suggestions are scrolled so the selected result is always visible.
* Verify all tests pass and all checks are green.

### Testing Instructions for Keyboard
Use the up and down arrows to navigate the suggestions as shown on the preview.

## Screenshots or screencast 

`URLInput:`
https://github.com/WordPress/gutenberg/assets/8436925/f941fc4e-43d8-415c-adc6-4fb5dddcfa40

`FormTokenField:`
https://github.com/WordPress/gutenberg/assets/8436925/b88a39a8-b4eb-4525-88c7-5dd5129bd745
